cherrypy/lib/cptools.py

```diff
@@ -410,14 +410,15 @@ def session_auth(**kwargs):
     return sa.run()
 
 
+keyword_args = ['* %s: %s' % (k, type(getattr(SessionAuth, k)).__name__)
+                for k in dir(SessionAuth) if not k.startswith('__')]
```

Comment:
Since you're refactoring this, you should probably do a generator expression instead of an in-memory list. Also, invent a better name for the var and use newer string formatting.

```suggestion
session_auth_subclasses = (
    '* {!s}: {!s}'.format(k, type(getattr(SessionAuth, k)).__name__)
    for k in dir(SessionAuth) if not k.startswith('__')
)
```

Though, I'd prefer that you didn't do code refactoring in a docs PR.

Reply:
I reverted the change in this section, but with the rest of the changes, the documentation should still be in better shape overall. Like you said, the code refactoring can be done in another pull request.